README
This archive contains the distribution Devel-IPerl,
version 0.012:

  Perl language kernel for Jupyter

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.025.


Changes
0.012 2022-09-14 00:33:57-0400

  Refactoring

   - Port from ZMQ::LibZMQ3 to ZMQ::FFI.
     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/113>.

0.011 2021-12-23 20:51:46-0500

  Features

   - Create a small Dockerfile for IPerl + PDL.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/57>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/108>.

  Bug fixes

   - Fix IPython version parsing in test (startstarz++).

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/110>.

0.010 2021-04-05 14:48:12-0500

  Features

   - Kernel now works with JupyterLab.

     Thanks to sknigh, Tmarinus, and vyuh for the bug report and to dlukes for
     the fixes.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/90>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/98>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/101>.

  Refactoring

   - Test console using Expect.pm.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/87>.

  Bug fixes

   - Update parsing the IPython version number in test (Joe Landman, joelandman++).

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/97>.

   - Update message protocol to use kernel-status messages
     (David Lukes, dlukes++).

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/98>.

   - Fix hanging when running notebook cells in rapid succession and when
     using "Run all cells" Thanks to David Lukes (dlukes++) for the report and
     t.fujiwara (tofjw++) for the fix.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/93>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/106>.

   - Unicode output streams on STDOUT/STDERR.

     Thanks to CH Su for the bug report and David Lukes (dlukes++) for the
     fix.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/35>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/92>,
     <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/98>.

   - Stop JSON decoding raw data blobs.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/100>

  Build changes

   - Test on more recent versions of Perl (Gabor Szabo, szabgab++).

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/99>.

   - Run tests on GitHub Actions CI.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/103>.

   - Make Devel::REPL a suggested dependency as Reply is the default backend.

     See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/104>.

0.009 2017-12-17 01:37:41-0600

  Feature
    - Installation on macOS and Windows are now supported. This is documented
      in `README.md`.
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/82>,
      <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/83>.
    - Align output style with IPython kernel (stphnlyd++).
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/66>.

  Fixes
    - Add explicit dependency on `B::Keywords` so that
      `Reply::Plugin::Autocomplete::Keywords` plugin loads.
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/53>.
    - Make sure more of the code base uses `jupyter` rather than `ipython`.
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/82>.

0.008 2017-09-08 02:47:34-0400
  Fixes
    - Hide IPerl namespace from PAUSE because a module with the name iPerl
      already exists on CPAN.

0.007 2017-09-07 23:46:03-0400
  Fixes
    - Fix ZMQ protocol to process multiple ZMQ identities by Tom McLaughlin
      (thomasjm++). See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/59>.
    - Use `jupyter` command to launch kernel rather than the deprecated
      `ipython` by Stephan Loyd (stphnlyd++). See
      <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/64>.
    - Build fixes for dzil by Mohammad S Anwar (manwar++).
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/77>.

  Enhancements
    - Add support for `is_complete_request` message type by Stephan Loyd
      (stphnlyd++). See <https://github.com/EntropyOrg/p5-Devel-IPerl/issues/62>,
      <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/63>.

  Refactoring
    - Split out ChartClicker plugin to separate distribution.
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/76>.
    - Split out PDLGraphicsGnuplot plugin to separate distribution.
      See <https://github.com/EntropyOrg/p5-Devel-IPerl/pull/75>.


0.006 2015-08-02 16:27:06-0500
  - add a Chart::Clicker plugin. Much thanks to
    Zhenyi Zhou (zhouzhen1++) for implementing this.

0.005 2015-05-02 20:08:27-0500
  - fix to how the kernel spec is written out. Generate the JSON in the
    bin/iperl script.
  - remove profile template in share/

0.004 2015-05-02 19:46:58-0500
  - update the IPython message spec to v5.0
  - support code completion for the Reply REPL backend
  - Add a kernel spec so that kernel switching can work in the IPython
    Notebook. This means that an IPython profile directory
    (~/.ipython/profile_perl/) is not needed to start a frontend.
  - add HMAC SHA256 signature to message spec

0.003 2015-03-29 13:04:14-0500
  - send warnings to frontend
  - add an environment variable $ENV{PERL_IPERL_IPYTHON} that indicates if
    IPerl is running under IPython.

0.002 2015-02-15 14:56:18-0600
  - support using Reply as the REPL backend
  - add Displayable role
  - add support for IPerl helpers
  - add plugin for PDL::Graphics::Gnuplot
  - add example of rich display system
  - convert `iperl` script from shell script to Perl
  - dev: Travis-CI builds

0.001 2014-08-16 18:27:47-0500
  - initial release
  - still needs more documentation and testing
  - release early, release often
  - happy CPAN Day!


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.025.
use strict;
use warnings;

use 5.013002;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "Perl language kernel for Jupyter",
  "AUTHOR" => "Zakariyya Mughal <zmughal\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.06"
  },
  "DISTNAME" => "Devel-IPerl",
  "EXE_FILES" => [
    "bin/iperl"
  ],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.013002",
  "NAME" => "Devel::IPerl",
  "PREREQ_PM" => {
    "B::Keywords" => 0,
    "Capture::Tiny" => 0,
    "Digest::SHA" => 0,
    "Env" => 0,
    "File::Copy::Recursive" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "FindBin" => 0,
    "Getopt::Long" => 0,
    "HTML::FromANSI" => 0,
    "IO::Async::Handle" => 0,
    "IO::Async::Loop" => 0,
    "IO::Async::Routine" => 0,
    "IO::Handle" => 0,
    "JSON::MaybeXS" => 0,
    "LWP::UserAgent" => 0,
    "List::AllUtils" => 0,
    "Log::Any" => 0,
    "MIME::Base64" => 0,
    "Markdown::Pod" => 0,
    "Moo" => 0,
    "Moo::Role" => 0,
    "MooX::HandlesVia" => 0,
    "MooX::Singleton" => 0,
    "MooX::Types::MooseLike::Base" => 0,
    "MooseX::HandlesConstructor" => 0,
    "Net::Async::ZMQ" => "0.002",
    "Net::Async::ZMQ::Socket" => 0,
    "PPI::Document" => 0,
    "Path::Class" => 0,
    "Reply" => 0,
    "Reply::Plugin" => 0,
    "Scalar::Util" => 0,
    "Try::Tiny" => 0,
    "UUID::Tiny" => 0,
    "ZMQ::FFI" => "1.18",
    "autodie" => 0,
    "base" => 0,
    "constant" => 0,
    "if" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "File::Which" => 0,
    "Test::More" => 0,
    "Test::Most" => 0,
    "version" => 0
  },
  "VERSION" => "0.012",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "B::Keywords" => 0,
  "Capture::Tiny" => 0,
  "Digest::SHA" => 0,
  "Env" => 0,
  "File::Copy::Recursive" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "File::Which" => 0,
  "FindBin" => 0,
  "Getopt::Long" => 0,
  "HTML::FromANSI" => 0,
  "IO::Async::Handle" => 0,
  "IO::Async::Loop" => 0,
  "IO::Async::Routine" => 0,
  "IO::Handle" => 0,
  "JSON::MaybeXS" => 0,
  "LWP::UserAgent" => 0,
  "List::AllUtils" => 0,
  "Log::Any" => 0,
  "MIME::Base64" => 0,
  "Markdown::Pod" => 0,
  "Moo" => 0,
  "Moo::Role" => 0,
  "MooX::HandlesVia" => 0,
  "MooX::Singleton" => 0,
  "MooX::Types::MooseLike::Base" => 0,
  "MooseX::HandlesConstructor" => 0,
  "Net::Async::ZMQ" => "0.002",
  "Net::Async::ZMQ::Socket" => 0,
  "PPI::Document" => 0,
  "Path::Class" => 0,
  "Reply" => 0,
  "Reply::Plugin" => 0,
  "Scalar::Util" => 0,
  "Test::More" => 0,
  "Test::Most" => 0,
  "Try::Tiny" => 0,
  "UUID::Tiny" => 0,
  "ZMQ::FFI" => "1.18",
  "autodie" => 0,
  "base" => 0,
  "constant" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

if ( $^O eq 'MSWin32' ) {
	$WriteMakefileArgs{PREREQ_PM}{'Alien::ZMQ::latest'} = $FallbackPrereqs{'Alien::ZMQ::latest'} = '0';
}

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
